search/includes/classes/commands/class-corecommand.php

```diff
@@ -233,13 +231,20 @@ public function index( $args, $assoc_args ) {
 					$sites = [ $blog_ids ];
 				}
 				foreach ( $sites as $site ) {
+					$valid_sites = get_sites(
+						[
+							'fields'   => 'ids',
+							'site__in' => $sites,
+							'number'   => count( $sites ),
+						]
+					);
 					// Verify it's a valid blog ID before proceeding.
-					if ( ! in_array( (int) $site, $all_sites, true ) ) {
+					if ( ! in_array( (int) $site, $valid_sites, true ) ) {
 						WP_CLI::error( "Blog ID {$site} does not exist!" );
 					}
 				}
```

Comment:
Does the get_sites() query have to take place inside the loop? Feels like something we would just need to run once right?